manager/controllers/app/fybrikapplication_controller.go

```diff
@@ -59,17 +62,34 @@ type FybrikApplicationReconciler struct {
 
 // Reconcile reconciles FybrikApplication CRD
 // It receives FybrikApplication CRD and selects the appropriate modules that will run
-// The outcome is a Plotter containing multiple Blueprints that run on different clusters
+// The outcome is a Plotter that describes the modules required, the clusters in which they should run, and the overall data flows.
 func (r *FybrikApplicationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	log := r.Log.WithValues("fybrikapplication", req.NamespacedName)
+	log := r.Log.With().Str("fybrikapplication", req.NamespacedName.String()).Logger()
+
 	// obtain FybrikApplication resource
 	applicationContext := &api.FybrikApplication{}
 	if err := r.Get(ctx, req.NamespacedName, applicationContext); err != nil {
-		log.V(0).Info("The reconciled object was not found")
+		// TODO - Should this be fatal rather than error level?
+		log.Error().Err(err).Str(logging.FYBRIKAPPUUID, "Unknown").Msg("The reconciled object was not found")
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
+
+	// Add the globally unique identifier for the FybrikApplication instance to the annotations for use in logging
+	// Only need to do this once, and not on every update to this resource
+	uuid := utils.GetFybrikApplicationUUID(applicationContext)
+	if applicationContext.GetGeneration() == 1 {
+		annotations := applicationContext.GetAnnotations()
+		if annotations == nil {
+			annotations = map[string]string{}
+			annotations[utils.FYBRIKAPPUUID] = uuid
+			applicationContext.SetAnnotations(annotations)
+		} else {
+			annotations[utils.FYBRIKAPPUUID] = utils.GetFybrikApplicationUUID(applicationContext)
+		}
+	}
+
```

Comment:
sure

Reply:
@simanadler , sorry zerolog has a slightly different API, see [Sub-loggers](https://github.com/rs/zerolog/blob/0d2f0be4ee595b0f227574d22310abb749e31319/log.go#L27),
or you can consider to use [zapr](https://github.com/go-logr/zapr), which is logr wrapper over logr. My point is to prevent adding Str(logging.FYBRIKAPPUUID, uuid)  to every call.

`import (
	"fmt"

	"github.com/go-logr/logr"
	"github.com/go-logr/zapr"
	"go.uber.org/zap"
)

func main() {
	var log logr.Logger

	zapLog, err := zap.NewProduction()
	if err != nil {
		panic(fmt.Sprintf("who watches the watchmen (%v)?", err))
	}
	log = zapr.NewLogger(zapLog)
	log = log.WithValues("FYBRIKAPPUUID", "myApp")
	log.Info("Logr in action!", "the answer", 42)
	log.Error(fmt.Errorf("myError"), "error test")
	log = log.WithCallDepth(1)
	log.Error(fmt.Errorf("myError2"), "error test 2")
}`
output:

> {"level":"info","ts":1638291742.615371,"caller":"pkg/test.go:21","msg":"Logr in action!","FYBRIKAPPUUID":"myApp","the answer":42}
{"level":"error","ts":1638291742.615975,"caller":"pkg/test.go:22","msg":"error test","FYBRIKAPPUUID":"myApp","error":"myError","stacktrace":"main.main\n\tC:/Users/ALEXEYROYTMAN/goWork/src/github.com/ibm/ovsdb-etcd/pkg/test.go:22\nruntime.main\n\tC:/Go/src/runtime/proc.go:225"}
{"level":"error","ts":1638291742.615975,"caller":"runtime/proc.go:225","msg":"error test 2","FYBRIKAPPUUID":"myApp","error":"myError2","stacktrace":"runtime.main\n\tC:/Go/src/runtime/proc.go:225"}

Reply:
@roytman I fixed so we don't add the uuid on every call.